### PR TITLE
(FACT-3135) Add support for yaml anchors in Facter 4

### PR DESCRIPTION
**Pull request by AriaXLi**

### CODE REVIEW

File: lib/facter/custom_facts/util/parser.rb [hunk 1]
```diff
@@ -119,9 +119,9 @@ def parse_results
           cont = content.gsub(TIME, '"\1"')
 
           if Gem::Version.new(Psych::VERSION) >= Gem::Version.new('3.1.0') # Ruby 2.6+
-            YAML.safe_load(cont, permitted_classes: [Date])
+            YAML.safe_load(cont, permitted_classes: [Date], aliases: true)
           else
-            YAML.safe_load(cont, [Date])
+            YAML.safe_load(cont, [Date], aliases: true)
```
Comment (joshcooper) [inline lib/facter/custom_facts/util/parser.rb:124]: On older ruby versions, we'll need to pass this parameter as a positional argument. Here's the older API signature https://ruby-doc.org/stdlib-2.5.0/libdoc/psych/rdoc/Psych.html#method-c-safe_load

Comment (AriaXLi) [reply lib/facter/custom_facts/util/parser.rb]: Ahh okay, thank you for catching that, I had no idea

**Comment by joshcooper** [comment]
Ah sorry forgot to mention it'd be good to add a unit test for this. For example, there's a fixture for dates in external yaml facts https://github.com/puppetlabs/facter/blob/main/spec/fixtures/external_fact_yaml_date and there's a unit test to verify we can load the external fact.

**Comment by AriaXLi** [comment]
> Ah sorry forgot to mention it'd be good to add a unit test for this. For example, there's a fixture for dates in external yaml facts https://github.com/puppetlabs/facter/blob/main/spec/fixtures/external_fact_yaml_date and there's a unit test to verify we can load the external fact.

Ah okay, I'll add that test too to this PR!

**Review by mhashizume — APPROVED** [review]

**Pull request merged**